### PR TITLE
Initial version of file_sorter

### PR DESCRIPTION
**Pull request by Agata-Gizinska**
The aim of this program is to organize files in the current directory according to its extension (file format). 

The program consists of:
- Directories dictionary, which assigns specified file formats into appropriate categories;
- File_Format_Dictionary, which maps file formats into appropriate directories;
- main function, which scans all files in the current directory, checks each file suffix with a dictionary of file formats and creates a new directory for specified file formats.

### CODE REVIEW

File: clean_my_dir.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        file_path = Path(entry)
+        file_format = file_path.suffix.lower()
+        if file_format in File_Format_Dictionary:
+            directory_path = Path(File_Format_Dictionary[file_format])
+            directory_path.mkdir(exist_ok=True)
+            file_path.rename(directory_path.joinpath(file_path))
+
+
+if __name__ == "__main__":
+    main()
```
Comment (czarnadzuma11) [inline clean_my_dir.py:66]: Target directory should be a parameter

**Review by czarnadzuma11 — CHANGES_REQUESTED** [review]

File: clean_my_dir.py (hunk 1 above)
Comment (Agata-Gizinska) [reply clean_my_dir.py:66]: done

File: README.md [hunk 2]
```diff
@@ -25,8 +27,18 @@ Categories:
 
 ## Usage
 
-Using the terminal change the current directory to the directory with the clean_my_dir.py file. Run command:
+Using the terminal run the program and provide a path to the directory as an
+argument:
 
-```python clean_my_dir.py```
+```python clean_my_dir.py '\path\to\directory'```
```
Comment (czarnadzuma11) [inline README.md]: "/path/to/directory"
use double quotes so shell can expand strings and use / unix path separators. Your script should be able to handle both

File: README.md [hunk 3]
```diff
@@ -25,8 +27,18 @@ Categories:
 
 ## Usage
 
-Using the terminal change the current directory to the directory with the clean_my_dir.py file. Run command:
+Using the terminal run the program and provide a path to the directory as an
+argument:
```
Comment (czarnadzuma11) [inline README.md]: an argument -> a parameter

File: README.md [hunk 4]
```diff
@@ -25,8 +27,18 @@ Categories:
...
 
-Enjoy an well-organized directory!
+### Example usage:
+
+In this use case we will sort files in the test_files directory attached to 
+this repository.
+
+Run the program using the terminal:
+
+```python clean_my_dir.py '.\test_files'```
```
Comment (czarnadzuma11) [inline README.md]: \ -> /

File: clean_my_dir.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+File_Format_Dictionary = {
+                          file_format: directory
+                          for directory, file_formats in Directories.items()
+                          for file_format in file_formats
+                          }
+
+
+def main():
+    """The main function of the program. It scans all files in the prompted
+    directory, checks each file suffix with a dictionary of file formats and
```
Comment (czarnadzuma11) [inline clean_my_dir.py]: prompted directory -> specified directory

File: clean_my_dir.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    creates a new directory for specified file formats."""
+
+    parser = argparse.ArgumentParser(description='Provide directory path')
```
Comment (czarnadzuma11) [inline clean_my_dir.py]: Description string describes the function of the program.

File: clean_my_dir.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    parser.add_argument('dir', type=str, help='Directory path')
+    args = parser.parse_args()
+
+    os.chdir(args.dir)
+    for entry in os.scandir():
```
Comment (czarnadzuma11) [inline clean_my_dir.py]: instead of chdir just pass args.dir to os.scandir

File: clean_my_dir.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        if entry.is_dir():
+            continue
+        file_path = Path(entry)
+        file_format = file_path.suffix.lower()
+        if file_format in File_Format_Dictionary:
+            directory_path = Path(File_Format_Dictionary[file_format])
```
Comment (czarnadzuma11) [inline clean_my_dir.py]: rename directory_path to dest_directory or such

File: clean_my_dir.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+File_Format_Dictionary = {
+                          file_format: directory
+                          for directory, file_formats in Directories.items()
+                          for file_format in file_formats
+                          }
+
+
+def main():
+    """The main function of the program. It scans all files in the prompted
```
Comment (czarnadzuma11) [inline clean_my_dir.py]: Turn this into description string for the program, also mention that it will not sort in subdirectories

**Review by czarnadzuma11 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Agata-Gizinska) [reply README.md]: ok

File: README.md (hunk 3 above)
Comment (Agata-Gizinska) [reply README.md]: ok

File: README.md (hunk 4 above)
Comment (Agata-Gizinska) [reply README.md]: ok

File: clean_my_dir.py (hunk 5 above)
Comment (Agata-Gizinska) [reply clean_my_dir.py]: ok

File: clean_my_dir.py (hunk 6 above)
Comment (Agata-Gizinska) [reply clean_my_dir.py]: noted, deleted

File: clean_my_dir.py (hunk 8 above)
Comment (Agata-Gizinska) [reply clean_my_dir.py]: ok

File: clean_my_dir.py [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    directory, checks each file suffix with a dictionary of file formats and
+    creates a new directory for specified file formats."""
+
+    parser = argparse.ArgumentParser(description='Provide directory path')
+    parser.add_argument('dir', type=str, help='Directory path')
```
Comment (czarnadzuma11) [inline clean_my_dir.py:46]: Update readme

**Review by czarnadzuma11 — CHANGES_REQUESTED** [review]

File: clean_my_dir.py (hunk 7 above)
Comment (Agata-Gizinska) [reply clean_my_dir.py]: ok

File: clean_my_dir.py (hunk 9 above)
Comment (Agata-Gizinska) [reply clean_my_dir.py]: ok

File: clean_my_dir.py (hunk 10 above)
Comment (Agata-Gizinska) [reply clean_my_dir.py:46]: ok

**Review by czarnadzuma11 — APPROVED** [review]

**Pull request merged**